checks.d/mesos_master.py

```diff
@@ -0,0 +1,246 @@
+        'master/event_queue_messages'                       : ('mesos.cluster.event_queue_messages', GAUGE),
+        'master/invalid_framework_to_executor_messages'     : ('mesos.cluster.invalid_framework_to_executor_messages', GAUGE),
+        'master/invalid_status_update_acknowledgements'     : ('mesos.cluster.invalid_status_update_acknowledgements', GAUGE),
+        'master/invalid_status_updates'                     : ('mesos.cluster.invalid_status_updates', GAUGE),
+        'master/valid_framework_to_executor_messages'       : ('mesos.cluster.valid_framework_to_executor_messages', GAUGE),
+        'master/valid_status_update_acknowledgements'       : ('mesos.cluster.valid_status_update_acknowledgements', GAUGE),
+        'master/valid_status_updates'                       : ('mesos.cluster.valid_status_updates', GAUGE),
+    }
+
+    def _timeout_event(self, url, timeout, aggregation_key):
+        self.event({
+            'timestamp': int(time.time()),
+            'event_type': 'http_check',
+            'msg_title': 'URL timeout',
+            'msg_text': '%s timed out after %s seconds.' % (url, timeout),
+            'aggregation_key': aggregation_key
+        })
+
+    def _status_code_event(self, url, r, aggregation_key):
+        self.event({
+            'timestamp': int(time.time()),
+            'event_type': 'http_check',
+            'msg_title': 'Invalid reponse code for %s' % url,
+            'msg_text': '%s returned a status of %s' % (url, r.status_code),
+            'aggregation_key': aggregation_key
+        })
+
+    def _get_json(self, url, timeout):
+        # Use a hash of the URL as an aggregation key
+        aggregation_key = md5(url).hexdigest()
+        tags = ["url:%s" % url]
+        msg = None
+        status = None
+        try:
+            r = requests.get(url, timeout=timeout)
+            if r.status_code != 200:
+                self._status_code_event(url, r, aggregation_key)
+                status = AgentCheck.CRITICAL
+                msg = "Got %s when hitting %s" % (r.status_code, url)
+            else:
+                status = AgentCheck.OK
+                msg = "Mesos master instance detected at %s " % url
+        except requests.exceptions.Timeout as e:
+            # If there's a timeout
+            self._timeout_event(url, timeout, aggregation_key)
+            msg = "%s seconds timeout when hitting %s" % (timeout, url)
+            status = AgentCheck.CRITICAL
+        except Exception as e:
+            msg = e.message
+            status = AgentCheck.CRITICAL
+        finally:
+            if self.SERVICE_CHECK_NEEDED:
+                self.service_check(self.SERVICE_CHECK_NAME, status, tags=tags,
+                                   message=msg)
+                self.SERVICE_CHECK_NEEDED = False
+            if status is AgentCheck.CRITICAL:
+                self.warning(msg)
+                return None
```

Comment:
Should we raise an Exception here and stop the execution of the check run here ?